pkg/handler/project/project.go

```diff
@@ -766,3 +766,44 @@ func (h *handler) assignMemberToProject(db *gorm.DB, projectID string, req Assig
 
 	return slot, http.StatusOK, nil
 }
+
+// Details godoc
+// @Summary Get details of a project
+// @Description Get details of a project
+// @Tags Project
+// @Accept  json
+// @Produce  json
+// @Param Authorization header string true "jwt token"
+// @Param id path string true "Project ID"
+// @Success 200 {object} view.ProjectListDataResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /projects/{id} [get]
+func (h *handler) Details(c *gin.Context) {
+	projectID := c.Param("id")
+	if projectID == "" || !model.IsUUIDFromString(projectID) {
+		c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, ErrInvalidProjectID, nil, ""))
+		return
+	}
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler": "project",
+		"method":  "AssignMember",
+		"id":      projectID,
+	})
+
+	project, err := h.store.Project.One(h.repo.DB(), projectID)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			l.Info("project not found")
+			c.JSON(http.StatusNotFound, view.CreateResponse[any](nil, nil, err, nil, ""))
```

Comment:
I think it should be more specific like "project not found" not "record not found"